axum/src/routing/mod.rs

```diff
@@ -212,6 +212,41 @@ where
     }
 
     /// Like [`nest`](Self::nest), but accepts an arbitrary `Service`.
+    ///
+    /// While [`nest`](Self::nest) requires [`Router`]s with the same type of
+    /// state, you can use this method to combine [`Router`]s with different
+    /// types of state:
+    ///
+    /// ```
+    /// use axum::{
+    ///     Router,
+    ///     routing::get,
+    ///     extract::State,
+    /// };
+    ///
+    /// #[derive(Clone)]
+    /// struct InnerState {}
+    ///
+    /// #[derive(Clone)]
+    /// struct OuterState {}
+    ///
+    /// async fn inner_handler(state: State<InnerState>) {}
+    ///
+    /// let inner_router = Router::new()
+    ///     .route("/bar", get(inner_handler))
+    ///     .with_state(InnerState {});
+    ///
+    /// async fn outer_handler(state: State<OuterState>) {}
+    ///
+    /// let app = Router::new()
+    ///     .route("/", get(outer_handler))
+    ///     .nest_service("/foo", inner_router)
+    ///     .with_state(OuterState {});
+    /// # let _: axum::routing::RouterService = app;
+    /// ```
+    ///
+    /// Note that the inner router will still inherit the fallback from the outer
+    /// router.
```

Comment:
I think this section should go on the docs for `nest`. When people try to use `nest` with different states I think they're most likely to read the docs for `nest` to see how to do that.

Reply:
I disagree. The docs should be where users need them and that is when using `nest`.

Reply:
I still disagree but this isn't a hill I'm willing to die on. So you win 😊

Reply:
The whole thing except the first sentence.